File: src/main/kotlin/dev/jefy/connectpro/user/domain/repository/UserRepository.java
package dev.jefy.connectpro.user.domain.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

import dev.jefy.connectpro.user.domain.model.User;
import dev.jefy.connectpro.user.domain.vo.Email;
import dev.jefy.connectpro.user.domain.vo.UserId;

/**
 * @author Jôph Yamba
 */
@Repository
public interface UserRepository extends JpaRepository<User, UserId> {
    Optional<User> findByEmail(Email email);

    boolean existsByEmail(Email email);
}

File: src/main/kotlin/dev/jefy/connectpro/user/domain/repository/UserRepository.kt
package dev.jefy.connectpro.user.domain.repository

import dev.jefy.connectpro.user.domain.model.User
import dev.jefy.connectpro.user.domain.vo.Email
import dev.jefy.connectpro.user.domain.vo.UserId
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, UserId> {
    fun findByEmail(email: Email): Optional<User>
    fun existsByEmail(email: Email): Boolean
}